src/support/global.c

```diff
@@ -47,62 +47,58 @@ __wt_endian_check(void)
 static void
 __global_calibrate_ticks(void)
 {
+	/*
+	 * Default to using __wt_epoch until we have a good value for the ratio.
+	 */
+	__wt_process.tsc_nsec_ratio = WT_TSC_DEFAULT_RATIO;
+	__wt_process.use_epochtime = true;
```

Comment:
@michaelcahill Did you read my reply on this particular line? This line turns off all rdtsc forever. We will never compute or otherwise use the tsc once this is true.

Reply:
@sueloverso, sorry if I misunderstood -- I changed it to start `true` but be set to `false` later in the function if we get a good sample.  Is there some reason why the function will exit in between?

Reply:
Oh, sorry, I see it now :-(

It's super confusing to have the `__wt_rdtsc` function not use `rdtsc`.  It would be much clearer to have the generic function called `__wt_gettime` or similar.

Reply:
The `wt_rdtsc` looks at the boolean to decide if it should use `wt_epoch` or use `rdtsc`. The calibrate function is calling `wt_rdtsc` so if the boolean is set, `wt_rdtsc` assumes we already decided to avoid rdtsc and calls . If we couldn't get good readings and compute a ratio we were falling back to setting the boolean and using `wt_epoch`.

Reply:
I'll fix the naming and possibly split the functions in WT-3825. And yes, you're correct that it will compute an incorrect ratio really close to 1.0 (my laptop shows 0.999995) but later uses will use rdtsc and compute incorrect times.